source/common/request_info/string_accessor_impl.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include "envoy/request_info/string_accessor.h"
+
+namespace Envoy {
+namespace RequestInfo {
+
+class StringAccessorImpl : public StringAccessor {
```

Comment:
How do you expect this to be used? I would imagine that dynamic metadata types would normally just extend StringAccessor and provide an asString() based on some internal data.

Reply:
Fair point, and if you'd like I'm happy to nuke it--it certainly isn't going to cause anyone grief to create the equivalent.  My thought was that for many such situations, the source of the data might not want to deal with the ownership/lifetime issues involved in owning the string data (or whatever is used to create the string data) and hence might simple want to hand over to DynamicMetadata an accessor that can be used for user-defined headers and forget about it.

Reply:
My inclination would be to leave it out, but let's see what other people think.